File: tools/pas2php-code-6/pas2php_include/p2pcontnrs.pas
{
 ***************************************************************************
 *                                                                         *
 *   This source is free software; you can redistribute it and/or modify   *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 *   This code is distributed in the hope that it will be useful, but      *
 *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU     *
 *   General Public License for more details.                              *
 *                                                                         *
 *   A copy of the GNU General Public License is available on the World    *
 *   Wide Web at <http://www.gnu.org/copyleft/gpl.html>. You can also      *
 *   obtain it by writing to the Free Software Foundation,                 *
 *   Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.        *
 *                                                                         *
 ***************************************************************************
}

unit P2PContnrs;

interface

uses SysUtils;

const
  A = True;

type

  CObjectList = class
  strict private
    FFreeObjects: boolean;
    FItems: array of TObject;
  protected
    function GetCount: integer;
    procedure SetCount(const ACount: integer);
  public
    constructor Create;
    destructor Destroy; override;
  public
    procedure Clear;
    function GetItem(const AIndex: integer): TObject;
    function IndexOf(const AObject: TObject): integer;
    procedure Add(const AObject: TObject);
    procedure Delete(AIndex: integer);
    procedure Remove(const AItem: TObject);
    procedure SetItem(const AIndex: integer; const AObject: TObject);
  published
    property Count: integer read GetCount write SetCount;
  end;

implementation

constructor CObjectList.Create;
begin
  inherited Create;
  FFreeObjects := True;
end;

destructor CObjectList.Destroy;
begin
  Clear;
  inherited Destroy;
end;

function CObjectList.GetCount: integer;
begin
  Result := Length(FItems);
end;

procedure CObjectList.SetCount(const ACount: integer);
begin
  SetLength(FItems, ACount);
end;

procedure CObjectList.Clear;
var
  LIndex: integer;
begin
  if FFreeObjects then begin
    for LIndex := High(FItems) downto 0 do begin
      FreeAndNil(FItems[LIndex]);
    end;
  end;
  SetLength(FItems, 0);
end;

function CObjectList.GetItem(const AIndex: integer): TObject;
begin
  Result := FItems[AIndex];
end;

procedure CObjectList.SetItem(const AIndex: integer; const AObject: TObject);
begin
  FItems[AIndex] := AObject;
end;

procedure CObjectList.Add(const AObject: TObject);
begin
  SetLength(FItems, Length(FItems) + 1);
  FItems[High(FItems)] := AObject;
end;

function CObjectList.IndexOf(const AObject: TObject): integer;
begin
  for Result := 0 to High(FItems) do begin
    if FItems[Result] = AObject then begin
      Exit;
    end;
  end;
  Result := -1;
end;

procedure CObjectList.Delete(AIndex: integer);
begin
  Assert((AIndex >= 0) and (AIndex < Length(FItems)));
  if FFreeObjects then begin
    FreeAndNil(FItems[AIndex]);
  end;
  while AIndex < High(FItems) do begin
    FItems[AIndex] := FItems[AIndex + 1];
    AIndex := AIndex + 1;
  end;
  SetLength(FItems, Length(FItems) - 1);
end;

procedure CObjectList.Remove(const AItem: TObject);
var
  LIndex: integer;
begin
  LIndex := IndexOf(AItem);
  if LIndex <> -1 then begin
    Delete(LIndex);
  end;
end;

end.

File: tools/pas2php-code-6/pas2php_include/p2pcontnrs.php
<?PHP
// 
// Converted From Pascal By Pas2Php v0.8
// Converts Object Pascal to PHP
// http://www.wascal.net/?page=pas2php
// Pas2Php v0.8 is Copyright (C) 2014-2016 Derek John Evans
// 
include_once('system.php');
include_once('sysutils.php');
// A = True;
define("A", true);
class CObjectList extends TObject {
  function __get($name) {
    // Count : Integer;
    if (strcasecmp($name, "Count") === 0) {
      return $this->GetCount();
    }
    return parent::__get($name);
  }
  function __set($name, $value) {
    // Count : Integer;
    if (strcasecmp($name, "Count") === 0) {
      return $this->SetCount($value);
    }
    return parent::__set($name, $value);
  }
  // FFreeObjects : Boolean;
  var $FFreeObjects = false;
  // FItems : Array of TObject;
  var $FItems = array();
  // function GetCount : Integer;
  function GetCount() {
    // Result : Integer;
    $Result = 0;
    $Result = Length($this->FItems);
    return $Result;
  }
  // procedure SetCount(const ACount: Integer);
  function SetCount($ACount) {
    SetLength($this->FItems, $ACount);
  }
  // constructor Create;
  function __construct() {
    parent::__construct();
    $this->FFreeObjects = true;
  }
  // destructor Destroy;  Override;
  function __destruct() {
    $this->Clear();
    parent::__destruct();
  }
  // procedure Clear;
  function Clear() {
    // LIndex : Integer;
    $LIndex = 0;
    if ($this->FFreeObjects) {
      for ($LIndex = High($this->FItems), $LIndex_High_ = 0; $LIndex >= $LIndex_High_; $LIndex--) {
        FreeAndNil($this->FItems[$LIndex]);
      }
    }
    SetLength($this->FItems, 0);
  }
  // function GetItem(const AIndex: Integer) : TObject;
  function GetItem($AIndex) {
    // Result : TObject;
    $Result = null;
    $Result = $this->FItems[$AIndex];
    return $Result;
  }
  // function IndexOf(const AObject: TObject) : Integer;
  function IndexOf($AObject) {
    // Result : Integer;
    $Result = 0;
    for ($Result = 0, $Result_High_ = High($this->FItems); $Result <= $Result_High_; $Result++) {
      if ($this->FItems[$Result] == $AObject) {
        return($Result);
      }
    }
    $Result =  - 1;
    return $Result;
  }
  // procedure Add(const AObject: TObject);
  function Add($AObject) {
    SetLength($this->FItems, Length($this->FItems) + 1);
    $this->FItems[High($this->FItems)] = $AObject;
  }
  // procedure Delete(AIndex: Integer);
  function Delete($AIndex) {
    Enforce($AIndex >= 0 && $AIndex < Length($this->FItems));
    if ($this->FFreeObjects) {
      FreeAndNil($this->FItems[$AIndex]);
    }
    while ($AIndex < High($this->FItems)) {
      $this->FItems[$AIndex] = $this->FItems[$AIndex + 1];
      $AIndex = $AIndex + 1;
    }
    SetLength($this->FItems, Length($this->FItems) - 1);
  }
  // procedure Remove(const AItem: TObject);
  function Remove($AItem) {
    // LIndex : Integer;
    $LIndex = 0;
    $LIndex = $this->IndexOf($AItem);
    if ($LIndex !=  - 1) {
      $this->Delete($LIndex);
    }
  }
  // procedure SetItem(const AIndex: Integer;const AObject: TObject);
  function SetItem($AIndex, $AObject) {
    $this->FItems[$AIndex] = $AObject;
  }
}
?>